ynput/OpenPype
# Ftrack: Hierarchical <> Non-Hierarchical attributes sync fix

## Changelog Description
Sync between hierarchical and non-hierarchical attributes should be fixed and work as expected. Action should sync the values as expected and event handler should do it too and only on newly created entities.

## Additional info
Action did not use proper table to get and change attribute values (user `ContextCustomAttributeValue` instead of `CustomAttributeValue`), so the received values were not right and changes were not propagated as expected. At the same time action had few bugs in way how values were pushed to ftrack. Event handler which should sync values automatically was based on task changes so it really never did changes as expected until a task was created/changed/moved/removed.

## Developer notes
Both handlers are pushing the value to task (which was original purpose of the handlers) but the logic should be probably split into 2 handlers or the logic of pushing to task should be enabled/disabled by settings.

Event handler does not handle cases when a parent is changing value. E.g. if Shot would inherit frameStart from Project and Project would change the value, the hierarchical value of frameStart would not be propagated to non-hierarchical on the shot -> handling of these usecases would make the logic much complicated and slower.

The event handler will set explicit value of hierarchical value all the time on change or create which may be confusing.

## Testing notes:
1. Run ftrack event server
2. Make sure settings for the action have some valid values `project_settings/ftrack/events/sync_hier_entity_attributes`
- Default values have `Shot` and `Asset Build` in types & `frameStart` and `frameEnd` in attributes
- Make sure the entity types have the attributes in ftrack
3. Go to ftrack project (you can create one for testing purposes)
4. Create new `Shot`
5. It's `frameStart` and `frameEnd` should be propagated from hierarchical values to non-hierarchical attribute
6. Change a value in hierarchical attribute and check if is propagated to non-hierarchical (and in reverse)
7. Create a `Task` under the shot
8. The task should have set the same attribute values as the shot in both hierarchical and non-hierarchical attributes
9. Create new entity which is not in settings (e.g. `Sequence`)
10. Entity hierarchical attribute values should not be propagated to non-hierarchical
11. Create Task under the entity
12. Created task hierarchical and non-hierarchical attribute values are untouched
